torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -1161,7 +1161,16 @@ def _shape_match(
         op: "cutlass_library.gemm_op.GemmOperation",  # type: ignore[name-defined]  # noqa: F821
     ) -> bool:
         X, W = self.input_nodes[0], self.input_nodes[1]
-        return X.get_size()[1] == W.get_size()[0]
+        X_size, W_size = X.get_size(), W.get_size()
+        if len(X_size) != len(W_size):
+            log.info("X and W have different ranks")
```

Comment:
```suggestion
            log.info("X and W have different ranks: %d, %d", len(X_size), len(W_size))
```
This would be effectively free if you cached the lens anyway?